File: Raven.Logger/Logger.cs
using Microsoft.Extensions.Logging;
using Raven.DB.PSQL.Entity;
using Raven.DB.PSQL.Logger;
using Raven.DB.Redis;
using System.Text.Json;

namespace Raven.Logger
{
    public class Logger
    {

        public static void Log(LogLevel logLevel, string message)
        {
            Logs log = new Logs()
            {
                LogLevel = logLevel,
                Message = message,
                DateTime = DateTime.UtcNow,
                LogSender = "Raven"
            };
            var db = new RedisLogsDbContext();
            db.WriteAsync(JsonSerializer.Serialize(log));
        }

        public static void FlushBuffer()
        {
            var dbLogs = new RedisLogsDbContext()
                .ReadAsync()
                .Result
                .ToList();
            var logs = dbLogs.Select(o => JsonSerializer.Deserialize<Logs>(o)).ToList();
            LogsHandler.ImportLogsAsync(logs);
            new RedisLogsDbContext().ClearAsync();
        }

    }
}

File: Raven.TagsPredictor/logger.py
import uuid

import orjson
from datetime import datetime

import redis

LOGS_KEY = "logs"

async def get_redis_connection():
    r = redis.Redis(host='redis-logs', port='6379')
    return r

async  def log(log_level, message):
    redis = await get_redis_connection()
    data = {"Id":str(uuid.UUID(int=0)),
            "LogSender": "Raven.TagsPredictor",
            "LogLevel":log_level,
            "DateTime":datetime.utcnow(),
            "Message":message
            }
    redis.rpush(LOGS_KEY, orjson.dumps(data).decode())